CMakeLists.txt

```diff
@@ -502,12 +502,23 @@ if(BUILD_DOCUMENTATION)
 
         file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/readthedocs_custom_template.cmake [=[
 
-            file(DOWNLOAD "https://fast-dds.docs.eprosima.com/_/downloads/en/v${PROJECT_VERSION}/htmlzip/" "./eprosima-fast-rtps.zip")
+            file(DOWNLOAD "https://fast-dds.docs.eprosima.com/_/downloads/en/v${PROJECT_VERSION}/htmlzip/" "./eprosima-fast-rtps.zip"
+                 STATUS DOWNLOAD_STATUS)
+            list(GET DOWNLOAD_STATUS 0 DOWNLOAD_RC)
+            
+            if (DOWNLOAD_RC)
+                message(STATUS "Unable to download documentation for version ${PROJECT_VERSION}. Falling back to master")
+                file(DOWNLOAD "https://fast-dds.docs.eprosima.com/_/downloads/en/master/htmlzip/" "./eprosima-fast-rtps.zip")
+            endif()
             # TODO: when windows ci CMake version surpasses 17 favor file() instead of UNZIP as in the next line
             # file(ARCHIVE_EXTRACT INPUT "./eprosima-fast-rtps.zip" DESTINATION  "${PROJECT_BINARY_DIR}/doc/")
             execute_process(COMMAND "${UNZIP_EXE}" "./eprosima-fast-rtps.zip" -d "${PROJECT_BINARY_DIR}/doc/")
             file(REMOVE_RECURSE "${PROJECT_BINARY_DIR}/doc/manual")
-            file(RENAME "${PROJECT_BINARY_DIR}/doc/eprosima-fast-rtps-v${PROJECT_VERSION}" "${PROJECT_BINARY_DIR}/doc/manual")
+            if (DOWNLOAD_RC)
+                file(RENAME "${PROJECT_BINARY_DIR}/doc/eprosima-fast-rtps-master" "${PROJECT_BINARY_DIR}/doc/manual")
+            else()
+                file(RENAME "${PROJECT_BINARY_DIR}/doc/eprosima-fast-rtps-v${PROJECT_VERSION}" "${PROJECT_BINARY_DIR}/doc/manual")
+            endif()
```

Comment:
This if-else can be substituted by simple statements if you create a local version var containing either `v${PROJECT_VERSION}` or `latest` depending on the downloaded zip

Reply:
I actually tried to implement this but the inline cmake script was not playing nice with the configure step when setting a variable via SET so I left it as is.